### PR TITLE
Disable dfs access based on the pool the delegate belongs to

### PR DESCRIPTION
**Pull request by fraibacas**

### CODE REVIEW

File: dfs/nfs/server.go [hunk 1]
```diff
@@ -149,13 +156,33 @@ func (c *Server) Clients() []string {
...
+func (c *Server) filterHostsWithoutDfsPerms(clients []string) []string {
+	if c.clientValidator == nil {
+		return clients
+	}
+	filteredClients := []string{}
+	for _, client := range clients {
+		if c.clientValidator.ValidateClient(client) {
+			filteredClients = append(filteredClients, client)
+		} else {
+			glog.Infof("Filtered NFS client with ip %s", client)
```
Comment (iancmcc) [inline dfs/nfs/server.go]: I would drop this log level.

File: cli/api/daemon.go [hunk 2]
```diff
@@ -727,7 +741,11 @@ func (d *daemon) startAgent() error {
 			"zkpath": poolPath,
 		}).Info("Established pool-based connection to ZooKeeper")
 
-		if options.NFSClient != "0" {
+		if delegateHasDFSAccess() {
+			log.Info("Did not mount the distributed filesystem. Delegate does not have DFS permissions")
```
Comment (iancmcc) [inline cli/api/daemon.go]: This should be Debug level

File: cli/api/daemon.go [hunk 3]
```diff
@@ -727,7 +741,11 @@ func (d *daemon) startAgent() error {
...
+		} else if options.NFSClient == "0" {
```
Comment (iancmcc) [inline cli/api/daemon.go]: This should be Debug level

File: facade/dfs.go [hunk 4]
```diff
@@ -646,3 +646,31 @@ func (info *registryVersionInfo) start(isvcsRoot string, hostPort string) (*dock
...
+}
+
+func NewDfsClientValidator(fac *Facade, ctx datastore.Context) DfsClientValidator {
+	return &clientValidator{ctx, fac}
+}
+
+func (val *clientValidator) ValidateClient(hostIP string) bool {
+	host, err := val.facade.GetHostByIP(val.context, hostIP)
+	if err != nil || host == nil {
+		glog.Errorf("Unable to load host with ip %s", hostIP)
```
Comment (iancmcc) [inline facade/dfs.go]: Should this be an Error? Error means we get a support ticket and the operator gets paged. Warning, maybe?

**Comment by fraibacas** [comment]
retest this

**Comment by fraibacas** [comment]
retest this

**Comment by fraibacas** [comment]
retest this

**Pull request merged**